### PR TITLE
test: Skipping one test in API Edit Spec for functional bugs

### PR DESCRIPTION
**Pull request by NandanAnantharamu**
cypress/e2e/Sanity/Datasources/RestApiOAuth2Validation_spec.ts --> there is a bug hence commenting this test
cypress/e2e/Regression/ServerSide/ApiTests/API_Edit_spec.js --> there is a bug hence commenting one test in this spec file

/ok-to-test tags="@tag.Sanity"

> [!TIP]
> 🟢 🟢 🟢 All cypress tests have passed! 🎉 🎉 🎉
> Workflow run: <https://github.com/appsmithorg/appsmith/actions/runs/12349492331>
> Commit: b7237c480acc2c974d124101c986caa13672049a
> <a href="https://internal.appsmith.com/app/cypress-dashboard/rundetails-65890b3c81d7400d08fa9ee5?branch=master&workflowId=12349492331&attempt=1" target="_blank">Cypress dashboard</a>.
> Tags: `@tag.Sanity`
> Spec:
> <hr>Mon, 16 Dec 2024 10:00:38 UTC

## Summary by CodeRabbit

- **Bug Fixes**
	- Updated the locator for the "settings" element to enhance precision.
	- Skipped a test case related to loading state after cancellation due to an existing bug.
	- Skipped a test case for creating an API with app URL due to an existing bug.

- **Tests**
	- Adjusted test case statuses to reflect current functionality and known issues.

### CODE REVIEW

File: app/client/cypress/locators/apiWidgetslocator.json [hunk 1]
```diff
@@ -47,7 +47,7 @@
   "page": ".single-select >div",
   "propertyList": ".binding",
   "actionlist": ".action div div",
-  "settings": "span:contains('Settings')",
+  "settings": "[data-testid='t--action-settings-trigger']",
```
Comment (sagar-qa007) [inline app/client/cypress/locators/apiWidgetslocator.json:50]: Do we need this change here? Looks like no need in this PR.

**Review by sagar-qa007 — CHANGES_REQUESTED** [review]

File: app/client/cypress/locators/apiWidgetslocator.json (hunk 1 above)
Comment (NandanAnantharamu) [reply app/client/cypress/locators/apiWidgetslocator.json:50]: yes this change is needed as there is better locator which we have updated and the earlier one was failing as well hence i had to update refer the dashbard link for the failure

**Review by sagar-qa007 — APPROVED** [review]

**Pull request merged**